File: src/main/java/burrows/apps/giphy/example/di/module/AppModule.java
package burrows.apps.giphy.example.di.module;


import android.app.Application;
import android.content.Context;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
@Module
public class AppModule {
  private Application application;

  public AppModule(final Application application) {
    this.application = application;
  }

  @Provides @Singleton Application provideApplication() {
    return application;
  }

  @Provides @Singleton Context provideApplicationContext() {
    return application.getApplicationContext();
  }
}

File: src/main/java/burrows/apps/example/gif/presentation/di/module/AppModule.kt
package burrows.apps.example.gif.presentation.di.module

import android.app.Application
import android.content.Context
import dagger.Binds
import dagger.Module

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
@Module(includes = [LeakCanaryModule::class])
abstract class AppModule {
  @Binds
  abstract fun providesContext(application: Application): Context
}
